code/datums/flock/flock.dm

```diff
@@ -358,18 +360,56 @@
 	for (var/mob/M in src.annotation_viewers)
 		M.client?.images -= I
 
+// naming
+
+/datum/flock/proc/pick_name(flock_type)
+	var/name
+	var/name_found = FALSE
+	var/tries = 0
+	var/max_tries = 5000 // really shouldn't occur
+
+	while (!name_found && tries < max_tries)
+		if (flock_type == "flock")
+			name = "[pick(consonants_lower)][pick(vowels_lower)].[pick(consonants_lower)][pick(vowels_lower)]"
+			if (!flocks[name])
+				name_found = TRUE
+		else if (flock_type == "flocktrace")
+			var/name_picked = FALSE
+			name = "[pick(consonants_upper)][pick(vowels_lower)].[pick(vowels_lower)]"
+			for (var/mob/living/intangible/flock/trace/T as anything in src.traces)
+				if (T.real_name == name)
+					name_picked = TRUE
+					break
+			name_found = !name_picked
+		else
+			if (flock_type == "flockdrone")
+				name = "[pick(consonants_lower)][pick(vowels_lower)].[pick(consonants_lower)][pick(vowels_lower)].[pick(consonants_lower)][pick(vowels_lower)]"
+			else if (flock_type == "flockbit")
+				name = "[pick(consonants_upper)].[rand(10,99)].[rand(10,99)]"
+			if (!src.active_unit_names[name])
+				name_found = TRUE
+				src.active_unit_names[name] = TRUE
+		tries++
+	if (!name_found && tries == max_tries)
+		logTheThing("debug", null, null, "Too many tries were reached in trying to name a flock or one of its units.")
+		return "error"
+	return name
+
 // UNITS
 
 /datum/flock/proc/registerUnit(var/atom/movable/D)
 	if(isflock(D))
 		src.units |= D
+		if (src.active_unit_names[D:real_name])
+			D:real_name = istype(D, /mob/living/critter/flock/drone) ? src.pick_name("flockdrone") : src.pick_name("flockbit")
```

Comment:
I'd rather the name be put into the list by `pick_name`.
Either way, get rid of `:`. There is already a type check you can cast and use dot.

Reply:
It already is put in by pick_name?

Reply:
it's indexed by real_name though, which is not guaranteed to be unique or static

Reply:
Sorry, just to clarify, this just repicks the flock unit's name if its name is already found in the units, otherwise no naming happens. Atm, all flock units are registered before their names are picked, except for one case which I accounted for. Was meant mainly for if wanting to undormantize a flock unit, thing for the future I think, but can be removed if wanted.

pick_name should guarantee it to be unique, not sure if possible to account for static name changes, but I think ideally a drone's real_name shouldn't be changing